### PR TITLE
[WFLY-12442]:Upgrade jboss-saaj-api and saaj-impl from 1.3 to 1.4

### PR DESCRIPTION
**Pull request by jimma**
….0.CR1
https://issues.jboss.org/browse/WFLY-12442

### CODE REVIEW

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 1]
```diff
@@ -7448,6 +7453,22 @@
...
+                    <exclusion>
+                        <groupId>javax.xml.stream</groupId>
+                        <artifactId>stax-api</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>javax.activation</groupId>
+                        <artifactId>activation</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
[the remark this answers is not in the document]
Comment (jimma) [reply pom.xml]: Fixed. Thanks.

**Comment by psakar** [comment]
@rgdoliveira this will require changes in test release + new project stream

**Comment by kabir** [comment]
@jimma A few of the jobs seem messed up, but for a bunch of the others there are 22 similar failures

**Comment by jimma** [comment]
@kabir  Looks we have to do more work to upgrade saaj-impl to 1.4. I am looking at these Ws test failures.

**Comment by scottmarlow** [comment]
@jimma please do let us know if you hit obstacles with SAAJ 1.4 that cannot be solved.

**Comment by bstansberry** [comment]
@jimma More exclusions needed:

```
[04:20:21][Step 2/3] [WARNING] Rule 0: org.apache.maven.plugins.enforcer.BanTransitiveDependencies failed with message:
[04:20:21][Step 2/3] org.wildfly:wildfly-feature-pack:pom:18.0.0.Beta1-SNAPSHOT
[04:20:21][Step 2/3]    org.jboss.spec.javax.xml.soap:jboss-saaj-api_1.4_spec:jar:1.0.0.CR1:compile has transitive dependencies:
[04:20:21][Step 2/3]       jakarta.activation:jakarta.activation-api:jar:1.2.1:compile
[04:20:21][Step 2/3] 
[04:20:21][Step 2/3] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M2:enforce (ban-transitive-deps) on project wildfly-feature-pack: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
```

**Comment by jimma** [comment]
@scottmarlow @bstansberry  It turns out that we can't upgrade to saaj 1.4 , even upgrade to 1.3.29.
There is one[ problematic commit]( https://github.com/jboss/saaj-impl/commit/148757613467ee3c60ac8db364c80cc842348713#diff-a1e80cc152d578329e259c567d2b9c96R94 )  in saaj 1.3.29 which changed saaj behavior to create one dom document instead of using the passed by element part's owner document.  This leads  webservice subystem complains many errors like : 
`javax.xml.ws.soap.SOAPFaultException: WRONG_DOCUMENT_ERR: A node is used in a different document than the one that created it.
`

I am going to look at what we can do in our saaj-impl 1.4 fork to fix these failures. It will probably take a lot more time than expected to get this work.

**Comment by jimma** [comment]
Retest this please

**Comment by bstansberry** [comment]
@darranl FYI re the 2nd commit

**Comment by jimma** [comment]
Looks the batch api upgrade in WFLY master broke enforcer rule :

```
[04:50:40][Step 2/3] [INFO] ------------------< org.wildfly:wildfly-batch-jberet >------------------
[04:50:40][Step 2/3] [INFO] Building WildFly: Batch Integration Subsystem (JBeret implementation) 1
[04:50:41][Step 2/3] [WARNING] 
[04:50:41][Step 2/3] Dependency convergence error for org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:2.0.0.CR1 paths to dependency are:
[04:50:41][Step 2/3] +-org.wildfly:wildfly-batch-jberet:18.0.0.Beta1-SNAPSHOT
[04:50:41][Step 2/3]   +-org.jboss.spec.javax.batch:jboss-batch-api_1.0_spec:2.0.0.CR2
[04:50:41][Step 2/3]     +-org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:2.0.0.CR1
[04:50:41][Step 2/3] and
[04:50:41][Step 2/3] +-org.wildfly:wildfly-batch-jberet:18.0.0.Beta1-SNAPSHOT
[04:50:41][Step 2/3]   +-org.wildfly.transaction:wildfly-transaction-client:1.1.7.Final
[04:50:41][Step 2/3]     +-org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:1.0.1.Final
[04:50:41][Step 2/3] 
[04:50:41][Step 2/3] [WARNING] Rule 0: org.apache.maven.plugins.enforcer.BannedDependencies failed with message:
[04:50:41][Step 2/3] Found Banned Dependency: org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:jar:2.0.0.CR1
[04:50:41][Step 2/3] Use 'mvn dependency:tree' to locate the source of the banned dependencies.
[04:50:41][Step 2/3] [WARNING] Rule 1: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
[04:50:41][Step 2/3] Failed while enforcing releasability. See above detailed error message.
[04:50:41][Step 2/3] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M2:enforce (ban-bad-dependencies) on project wildfly-batch-jberet: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
[04:50:41][Step 2/3] [INFO] 
```

**Comment by bstansberry** [comment]
retest this please

@jimma We just merged the fix for the enforcer problem.

**Comment by jimma** [comment]
Retest this please

**Comment by jimma** [comment]
The failures after security manager is enabled is caused by https://issues.apache.org/jira/browse/WSS-654

**Comment by jamezp** [comment]
@jimma This needs a rebase and likely should be bumped to the 1.0.0.Final version.

**Comment by bstansberry** [comment]
I resolve the conflict via the github UI as I want to see what CI does.

**Comment by jimma** [comment]
@bstansberry  @jamezp  I already rebased and upgraded saaj 1.4 api to 1.0.0.Final.

**Comment by jimma** [comment]
Looks one failure from Linux JDK11 build is not relevant.

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
@jimma @jamezp @asoldano I just kicked off a retest of this as it's been a couple days, but once that's done, as per an earlier discussion we should merge this and deal with the picketlink issue later, just being sure that's sorted or this is reverted before WF 18 Final.  There's value in getting this in and tested in other testsuites that consume WF master.

**Review by jamezp — APPROVED** [review]

**Review by asoldano — APPROVED** [review]
Thanks Brian, clearly I agree on the approach.

**Pull request merged**